hivemq4/k8s-image/Dockerfile

```diff
@@ -1,46 +1,41 @@
 ARG BASEIMAGE=hivemq/hivemq4:dns-latest
+
+
+# Additional build image to unpack the zip files and change the permissions without retaining large layers just for those operations
+FROM alpine:3.19@sha256:c5b1261d6d3e43071626931fc004f70149baeba2c8ec672bd4f27761f8e1ad6b as unpack
+
 ARG PROMETHEUS_EXTENSION_VERSION=4.0.8
 
-FROM alpine:3.19@sha256:c5b1261d6d3e43071626931fc004f70149baeba2c8ec672bd4f27761f8e1ad6b as BUILD
-ARG PROMETHEUS_EXTENSION_VERSION
+RUN mkdir -p /opt/hivemq/extensions
 
-WORKDIR /opt/hivemq/extensions
+COPY hivemq-k8s-sync-extension-*.zip /tmp/hivemq-k8s-sync-extension.zip
+RUN unzip /tmp/hivemq-k8s-sync-extension.zip -d /opt/hivemq/extensions \
+    && rm /tmp/hivemq-k8s-sync-extension.zip \
+    && chmod -R 770 /opt/hivemq/extensions/hivemq-k8s-sync-extension
```

Comment:
Usually I am also very conservative with these type of changes.
Actually, the permissions were already changed from 775 to 770 a year ago shortly before a release and I reverted the changes because we did not have automated tests for them. Now I feel a bit more confident to make these changes.

In general, I want to change the permissions anyhow in a followup (maybe when migrating to gradle-oci, maybe afterwards), because the current permissions don't really make sense. For example, everything being executable and writable, does not make sense. Also, all the files can be world readable without any issue. But all this is something for a followup PR.

Reply:
Totally agree on fix/set the right permissions. Just a little worried about any changes on this image as there might be customers extending this image to create their own custom image. I guess we should keep them aware of any potential breaking change.

Let me give it a quick spin with the legacy operator using this image to check it works fine with root/non-root.